rust/src/lib.rs

```diff
@@ -327,248 +312,233 @@ pub extern "C" fn zcashlc_init_accounts_table_with_keys(
 ///   when you are done using it.
 #[no_mangle]
-pub extern "C" fn zcashlc_derive_extended_full_viewing_keys(
-    seed: *const u8,
-    seed_len: usize,
-    accounts: i32,
+pub extern "C" fn zcashlc_usk_to_ufvk(
+    usk_ptr: *const u8,
+    usk_len: usize,
     network_id: u32,
-) -> *mut FFIEncodedKeys {
+) -> *mut c_char {
     let res = catch_panic(|| {
         let network = parse_network(network_id)?;
-        let seed = unsafe { slice::from_raw_parts(seed, seed_len) };
-        let accounts = if accounts > 0 {
-            accounts as u32
-        } else {
-            return Err(format_err!("accounts argument must be greater than zero"));
-        };
-
-        Ok(FFIEncodedKeys::ptr_from_vec(
-            (0..accounts)
-                .map(|account| {
-                    let account = AccountId::from(account);
-                    let extfvk = ExtendedFullViewingKey::from(&sapling::spending_key(
-                        seed,
-                        network.coin_type(),
-                        account,
-                    ));
-                    let encoded = encode_extended_full_viewing_key(
-                        network.hrp_sapling_extended_full_viewing_key(),
-                        &extfvk,
-                    );
-                    FFIEncodedKey::new(account, encoded)
-                })
-                .collect(),
-        ))
+        unsafe { decode_usk(usk_ptr, usk_len) }.map(|usk| {
+            let ufvk = usk.to_unified_full_viewing_key();
+            CString::new(ufvk.encode(&network)).unwrap().into_raw()
+        })
     });
     unwrap_exc_or_null(res)
 }
 
-/// Derives a unified address from the given seed and account index.
+/// Returns the default external unified address for the given unified spending key.
 ///
 /// Returns the Bech32-encoded string representation of the derived address.
 ///
 /// # Safety
 ///
-/// - `seed` must be non-null and valid for reads for `seed_len` bytes, and it must have an
-///   alignment of `1`.
-/// - The memory referenced by `seed` must not be mutated for the duration of the function call.
-/// - The total size `seed_len` must be no larger than `isize::MAX`. See the safety documentation
+/// - `usk_ptr` must be non-null and must point to an array of `usk_len` bytes containing a unified
+///   spending key encoded as returned from the `zcashlc_init_accounts_table` or
+///   `zcashlc_derive_univied_spending_key` functions.
+/// - The memory referenced by `usk_ptr` must not be mutated for the duration of the function call.
+/// - The total size `usk_len` must be no larger than `isize::MAX`. See the safety documentation
 ///   of pointer::offset.
 /// - Call [`zcashlc_string_free`] to free the memory associated with the returned pointer
 ///   when done using it.
 #[no_mangle]
-pub extern "C" fn zcashlc_derive_unified_address_from_seed(
-    seed: *const u8,
-    seed_len: usize,
-    account_index: i32,
+pub extern "C" fn zcashlc_default_unified_address_from_spending_key(
```

Comment:
You had mention on a PR that it would be better to actually have the client of the API pass a diversifier index.
wouldn't it be better to do it now?

https://github.com/zcash/ZcashLightClientKit/issues/508

Reply:
As we discussed offline, requiring the caller to manage diversifier indices adds some potential foot-guns, so now the two new methods `zcashlc_get_current_address` and `zcashlc_get_next_available_address` handle everything.